topics/lua-api.md

```diff
@@ -375,22 +375,80 @@ This function triggers a breakpoint when using the Redis Lua debugger](/topics/l
 
 This function prints its argument in the [Redis Lua debugger](/topics/ldb) console.
 
-### <a name="redis.register_function()"></a> `redis.register_function(name, callback)`
+### <a name="redis.register_function()"></a> `redis.register_function`
 
 * Since version: 7.0.0
 * Available in scripts: no
 * Available in functions: yes
 
 This function is only available from the context of the `FUNCTION LOAD` command.
 When called, it registers a function to the loaded library.
-The function expects two input arguments: the function's registered name and its callback.
+The function can be called either with positional argument or with named arguments.
 
-Usage example:
+#### <a name="redis.register_function_pos_args"></a> positional arguments: `redis.register_functio(name, callback)`
+
+The first argument to `redis.register_functio` is a Lua string representing the function name.
+The second argument to `redis.register_functio` is a Lua function.
+
+Error raise on the following cases:
+
+* Wrong number of arguments
+* Wrong argument type
+* Another function with the same name already registered on the current library
```

Comment:
i don't think we need to go to that level of detail (considering the user gets a sensible error when it happens)